src/main/java/io/jenkins/plugins/audit/listeners/UserCreationListener.java

```diff
@@ -0,0 +1,86 @@
+package io.jenkins.plugins.audit.listeners;
+
+
+import hudson.Extension;
+
+import hudson.ExtensionList;
+import hudson.Extension;
+import io.jenkins.plugins.audit.RequestContext;
+import jenkins.security.SecurityListener;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.time.Instant;
+
+import io.jenkins.plugins.audit.event.CreateUser;
+import org.acegisecurity.userdetails.User;
+import org.apache.logging.log4j.audit.LogEventFactory;
+import org.acegisecurity.userdetails.UserDetails;
+
+/**
+ * Listener notified of user creation events.
+ */
+@Extension
+public class UserCreationListener extends SecurityListener {
+
+    /**
+     * Fired when a user was successfully authenticated using credentials. It could be password or any other credentials.
+     * This might be via the web UI, or via REST (using API token or Basic), or CLI (remoting, auth, ssh)
+     * or any other way plugins can propose.
+     *
+     * @param details details of the newly authenticated user, such as name and groups.
+     */
+    protected void authenticated(@Nonnull UserDetails details) {}
+
+    /**
+     * Fired when a user has failed to log in.
+     * Would be called after {@link #failedToAuthenticate}.
+     *
+     * @param username the user
+     */
+    protected void failedToLogIn(@Nonnull String username) {}
+
+    /**
+     * Fired when a user tried to authenticate but failed.
+     * UserLogListener does not use this abstract class method, however, rather than leaving
+     * its implementation empty it is made to throw an exception if the method is ever called.
```

Comment:
Am I missing something? I don't see where it is throwing an exception.

Reply:
those non-used methods were removed. see @jvz third comment above.